src/tests/integration/test_foundry_prove.py

```diff
@@ -327,23 +328,21 @@ def test_foundry_remove_node(
     assert_pass(test, prove_res)
 
 
-def assert_pass(test: str, prove_res: dict[tuple[str, int], tuple[bool, list[str] | None]]) -> None:
-    id = id_for_test(test, prove_res)
-    passed, log = prove_res[(test, id)]
-    if not passed:
-        assert log
-        pytest.fail('\n'.join(log))
+def assert_pass(test: str, prove_res: list[Proof]) -> None:
+    proof = single([proof for proof in prove_res if proof.id.split(':')[0] == test])
```

Comment:
Consider

```suggestion
def assert_pass(proof: Proof) -> None:
```

and checking for a single proof in the caller.

Reply:
Done.